nixos/modules/services/networking/wireguard.nix

```diff
@@ -91,7 +105,21 @@ let
         example = "rVXs/Ni9tu3oDBLS4hOyAUAa1qTWVA3loR8eL20os3I=";
         type = with types; nullOr str;
         description = ''
-          base64 preshared key generated by wg genpsk. Optional,
+          DEPRECATED BECAUSE IT PUTS A WORLD-READABLE COPY OF THE KEY IN /nix/store!
```

Comment:
For some people, this might not be a problem. Rather than depreciating this, I'd keep both supported, but make them mutually exclusive.